Source/NSPathCell.m

```diff
@@ -0,0 +1,357 @@
+}
+
+- (NSRect)rectOfPathComponentCell:(NSPathComponentCell *)cell 
+                        withFrame:(NSRect)frame 
+                           inView:(NSView *)view
+{
+  NSUInteger index = [_pathComponentCells indexOfObject: cell];
+  CGFloat cellWidth = (frame.size.width / (CGFloat)[_pathComponentCells count]);
+  return NSMakeRect(frame.origin.x + (cellWidth * (CGFloat)index),
+                    frame.origin.y,
+                    cellWidth,
+                    frame.size.height);
+}
+
+- (NSPathComponentCell *)pathComponentCellAtPoint:(NSPoint)point 
+                                        withFrame:(NSRect)frame 
+                                           inView:(NSView *)view
+{
+  NSUInteger c = [_pathComponentCells count];
+  NSUInteger woc = frame.size.width / c;
+  NSUInteger item = (NSUInteger)point.x / woc;
+
+  return [_pathComponentCells objectAtIndex: item];
+}
+
+- (NSPathComponentCell *) clickedPathComponentCell
+{
+  return _clickedPathComponentCell;
+}
+
+- (NSArray *) pathComponentCells
+{
+  return _pathComponentCells;
+}
+
+- (void) setPathComponentCells: (NSArray *)cells
+{
+  ASSIGNCOPY(_pathComponentCells, cells);
+}
+
+- (SEL) doubleAction
+{
+  return _doubleAction;
+}
+
+- (void) setDoubleAction: (SEL)action
+{
+  _doubleAction = action;
+}
+
+- (NSURL *) URL
+{
+  return [self objectValue];
+}
+
+- (void) setURL: (NSURL *)url
+{
+  [self setObjectValue: url];
+  [self setPathComponentCells: [NSPathCell _generateCellsForURL: url]];
+}
+
+- (id<NSPathCellDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setDelegate: (id<NSPathCellDelegate>)delegate
+{
+  _delegate = delegate;
+}
+
+- (void) drawInteriorWithFrame: (NSRect)frame inView: (NSView *)controlView
+{
+  NSUInteger count = [_pathComponentCells count];
+
+  [super drawInteriorWithFrame: frame
+                        inView: controlView];
+  if (count > 0)
+    {
+      NSEnumerator *en = [_pathComponentCells objectEnumerator];
+      NSPathComponentCell *cell = nil;
+      
+      while((cell = (NSPathComponentCell *)[en nextObject]) != nil)
+        {
+          NSRect f = [self rectOfPathComponentCell: cell
+                                         withFrame: frame
+                                            inView: controlView]; 
+          [cell drawInteriorWithFrame: f
+                               inView: controlView];
+        }
+    }
+}
+
+- (id) initWithCoder: (NSCoder *)coder
+{
+  self = [super initWithCoder: coder];
+  if ([coder allowsKeyedCoding])
+    {
+      [self setPathStyle: NSPathStyleStandard];
+      
+      if ([coder containsValueForKey: @"NSPathStyle"]) 
+        {
+          [self setPathStyle: [coder decodeIntegerForKey: @"NSPathStyle"]];
+        }
+      
+      if ([coder containsValueForKey: @"NSPathComponentCells"])
+        {
+          [self setPathComponentCells: [coder decodeObjectForKey: @"NSPathComponentCells"]];
+        }
+
+      if ([coder containsValueForKey: @"NSContents"])
```

Comment:
This already gets decoded by NSCell.

Reply:
RIGHT... but it's not getting decoded properly by NSCell. :)   That's the problem.

Reply:
Could you just write a locale setObjectValue: method and log the parameter you get handed in? It should be the URL.

Reply:
Do you have any clue as to why?

Reply:
I tried that, I am working on day job stuff now.   I will return to this in a few hours.

Reply:
fixed... I also figured out why it is implemented separately.  If we setObjectValue on NSPathComponentCell, it will show because it's parent is NSTextField...  we don't want this.  So it's implemented there separately.

I also got the NSContents stuff working.  I am going to commit this and then clean up the path components stuff.  I am tempted to keep it only because that key IS used in the nib stuff.   It might be useful.